File: gitea_repos_master/whatsapp/whatsapp-baileys-api/src/logic/sentry/sentry.ts
import * as Sentry from '@sentry/node';
import logger from '../logger/winston';

export default class SentryConfig {
  constructor(sentry_url: string, mode: string) {
    if (sentry_url && mode === 'prod') {
      Sentry.init({
        dsn: sentry_url,
        environment: mode,
        tracesSampleRate: 1.0
      });
      logger.info('Sentry is active');
      return;
    }
    logger.info('Sentry is disabled');
  }
}

File: gitea_repos_master/isphere-services/captcha-image-api/src/logic/sentry/sentry.py
import sentry_sdk

from src.common.utils import SingletonLogging


class Sentry(SingletonLogging):
    IGNORE_EXCEPTIONS = [KeyboardInterrupt]

    def __before_send(self, event, hint):
        if "exc_info" in hint:
            _, exc_value, _ = hint["exc_info"]
            for exception in Sentry.IGNORE_EXCEPTIONS:
                if isinstance(exc_value, exception):
                    return None
        return event

    def create(self, dsn: str, mode: str):
        if mode != "prod":
            self.logger.info(f"Sentry MODE: '{mode}', GlitchTip disabled.")
            return
        self.logger.info(f"Sentry MODE: '{mode}', GlitchTip enabled.")
        sentry_sdk.init(
            dsn=dsn,
            before_send=self.__before_send,
            max_breadcrumbs=50,
            environment=mode,
            attach_stacktrace=True,
        )
